crates/cheatcodes/src/inspector.rs

```diff
@@ -1001,18 +1014,27 @@ where {
                     let account =
                         ecx.journaled_state.state().get_mut(&broadcast.new_origin).unwrap();
 
+                    let mut tx_req = TransactionRequest {
+                        from: Some(broadcast.new_origin),
+                        to: Some(TxKind::from(Some(call.target_address))),
+                        value: call.transfer_value(),
+                        input: TransactionInput::new(call.input.clone()),
+                        nonce: Some(account.info.nonce),
+                        chain_id: Some(ecx.env.cfg.chain_id),
+                        gas: if is_fixed_gas_limit { Some(call.gas_limit) } else { None },
+                        ..Default::default()
+                    };
+
+                    if let Some(auth_list) =
+                        self.active_delegation.and_then(|addr| self.delegations.remove(&addr))
+                    {
+                        tx_req.authorization_list = Some(vec![auth_list]);
+                        tx_req.transaction_type = Some(4); // EIP-7702
```

Comment:
there's no need to set tx type here I believe it would be correctly resolved on its own when building tx envelope

Reply:
Removed, thanks.